app_test.go

```diff
@@ -473,3 +491,76 @@ func TestNopLogger(t *testing.T) {
 	app := fxtest.New(t, NopLogger)
 	app.RequireStart().RequireStop()
 }
+
+type testErrorWithGraph struct {
+	graph string
+}
+
+func (we testErrorWithGraph) Graph() DotGraph {
+	return DotGraph(we.graph)
+}
+
+func (we testErrorWithGraph) Error() string {
+	return "great sadness"
+}
+
+func TestVisualizeError(t *testing.T) {
+	t.Run("NotWrappedError", func(t *testing.T) {
+		graph, err := VisualizeError(errors.New("great sadness"))
+		assert.Empty(t, graph)
```

Comment:
we don't need to assert that the graph is empty as long as error is not nil.

The contract for functions that return an error is that if err is not nil, the
other value is meaningless/can be anything.